src/js/components/Form/__tests__/Form-test-uncontrolled.js

```diff
@@ -501,6 +504,29 @@ describe('Form uncontrolled', () => {
               placeholder="email"
             />
           </FormField>
+          <FormField
+            label="Size"
+            name="test-select"
+            htmlFor="test-select__input"
```

Comment:
Why are we having to use the "__input" suffix? That seems like knowing too much about the internals. I would have expected this to be `htmlFor="text-select"` to match with the `id` of the contained Select.

Reply:
Would probably need to look into this as a separate fix since this is existing behavior, but in order for the label to associate with the input (using WAVE browser plugin to check this), we currently have to add `__input`. Agreed grommet should probably handle this internally.

Reply:
I think this is due to the fact that we internally manipulate the `id` passed to select input to be `${id}__input`, so we would probably want to to the same adjustment to `htmlFor` if the component is a Select.

Reply:
Something seems broken if the `id` isn't associated with the underlying `<input />` element, since that's what `htmlFor` is typically used for. For instance, CheckBox puts the `id` property on the underlying `<input type="checkbox" />` element.

It looks like Select puts the `id` on the DropButton. I can see where this might seem a good choice, but I think the input element is a better place. However, this would be a breaking change. Sigh. I wonder if we should support: `id={{ input: ..., button: ... }}`? It looks crazy, but is at least semantic. Hmm ...

Reply:
Yes, it would be a breaking change to move the placement of the input. We don't need to have the `__input` for the storybook example to work, I just wanted it to pass the WAVE check.

How would you feel about handling this work in a separate PR and having a little more discussion around implementation? I can remove the `__input` for now from the story for clarity if that'd be better.

Reply:
I'm fine taking care of the `id` element mapping in a separate PR. I would remove the `__input` from this PR.

Reply:
Done.